app/client/templates/Register.tsx

```diff
@@ -0,0 +1,195 @@
+import { Layout } from "@client/components/Layout";
+import { PageTitle } from "@client/components/PageTitle";
+import { Alert } from "@client/components/alert/Alert";
+import { CenteredFormContainer } from "@client/components/containers/CenteredFormContainer";
+import { BasicFormField } from "@client/components/forms/BasicFormField";
+import { BasicFormWithCustomFields } from "@client/components/forms/BasicFormWithCustomFields";
+import { AccomodationFieldGroup } from "@client/components/register/AccomodationFieldGroup";
+import { CostSummary } from "@client/components/register/PriceSummary";
+import { reverse, templates, useForm } from "@reactivated";
+import React from "react";
+
+export const Template = (props: templates.Register) => {
+  const form = useForm({ form: props.form });
+
+  const pageTitle = props.is_user_already_registered
+    ? "Update preferences"
+    : "Register";
+
+  const accomodationCheckboxesGroups = [
+    {
+      dinner: form.fields.dinner_day_1,
+      accomodation: form.fields.accommodation_day_1,
+      breakfast: form.fields.breakfast_day_2,
+    },
+    {
+      dinner: form.fields.dinner_day_2,
+      accomodation: form.fields.accommodation_day_2,
+      breakfast: form.fields.breakfast_day_3,
+    },
+    {
+      dinner: form.fields.dinner_day_3,
+      accomodation: form.fields.accommodation_day_3,
+      breakfast: form.fields.breakfast_day_4,
+    },
+  ];
+
+  const calculateAccomodationDayCost = (
+    breakfast: boolean,
+    dinner: boolean,
+    accomodation: boolean,
+  ) => {
+    if (!accomodation && !dinner && !breakfast) {
+      return 0;
+    }
+
+    if (dinner && breakfast) {
+      return props.zosia.price_whole_day;
+    }
+
+    if (dinner && !breakfast) {
+      return props.zosia.price_accommodation_dinner;
+    }
+
+    if (!dinner && breakfast) {
+      return props.zosia.price_accommodation_breakfast;
+    }
+
+    return props.zosia.price_accommodation;
+  };
+
+  const calculateDiscountForGroup = (accomodation: boolean) => {
+    return accomodation ? props.discount : 0;
+  };
+
+  const [accomodationCostPerGroup, setAccomodationCostPerGroup] =
+    React.useState(
+      accomodationCheckboxesGroups.map(({ dinner, accomodation, breakfast }) =>
```

Comment:
That looks wrong; why is this a state and not a simple variable?

Reply:
Refactored, look comment below.